GliaWidgets/SecureConversations/SecureConversations.FileUploadList.swift

```diff
@@ -0,0 +1,965 @@
+import UIKit
+#if canImport(QuickLookThumbnailing)
+  import QuickLookThumbnailing
+#endif
+
+private extension Int {
+    static let maxUnscrollableViewsOnDefaultContentSizeCategory = 3
+    static let maxUnscrollableViewsOnLargeContentSizeCategory = 2
+}
+
+extension SecureConversations {
+    class FileUploadListView: UIView {
+        struct Props: Equatable {
+            let maxUnscrollableViews: Int
+            let style: FileUploadListStyle
+            let uploads: IdCollection<FileUploadView.Props.Identifier, FileUploadView.Props>
+        }
+
+        var props: Props = .init(
+            maxUnscrollableViews: 2,
+            style: .init(item: .initial),
+            uploads: .init()
+        ) {
+            didSet {
+                renderProps()
+            }
+        }
+
+        private var uploadViews: [FileUploadView] {
+            return stackView.arrangedSubviews.compactMap { $0 as? FileUploadView }
+        }
+
+        private let scrollView = UIScrollView()
+        private let stackView = UIStackView()
+        private var heightLayoutConstraint: NSLayoutConstraint!
+
+        var cachedViews = IdCollection<FileUploadView.Props.Identifier, FileUploadView>()
+
+        init() {
+            super.init(frame: .zero)
+            setup()
+            layout()
+        }
+
+        @available(*, unavailable)
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+
+        func addUploadView(_ uploadView: FileUploadView) {
+            stackView.insertArrangedSubview(uploadView, at: 0)
+            updateHeight()
+            addAccessibilityProperties(for: uploadView)
+        }
+
+        func removeUploadView(_ uploadView: FileUploadView) {
+            stackView.removeArrangedSubview(uploadView)
+            uploadView.removeFromSuperview()
+            updateHeight()
+            removeAccessibilityProperties(for: uploadView)
+        }
+
+        func removeAllUploadViews() {
+            stackView.removeArrangedSubviews()
+            removeAccessibilityPropertiesForAllUploadViews()
+            updateHeight()
+        }
+
+        private func setup() {
+            stackView.axis = .vertical
+            // Assign empty array, because `accessibilityElements` is nil initially,
+            // and we need to append to/remove from it when views
+            // are added to/removed from stack view
+            stackView.accessibilityElements = []
+        }
+
+        private func layout() {
+            heightLayoutConstraint = autoSetDimension(.height, toSize: 0)
```

Comment:
let's minimize using PureLayout and start using native approach or our own helper-functions.